### PR TITLE
Fix forgot password, email verification

### PR DESCRIPTION
**Pull request by m-dotcom**

### CODE REVIEW

File: application/controllers/User.php [hunk 1]
```diff
@@ -373,17 +380,44 @@ public function register()
...
+                $this->session->set_flashdata("message", "Your account has been successfully created, please activate your account!");
+                redirect('/login', 'refresh');
+            } else {
+                $this->data['message'] = (validation_errors() ? validation_errors() : ($this->ion_auth->errors() ? $this->ion_auth->errors() : $this->session->flashdata('message')));
+                $this->data["post"] = $this->input->post();
+                $this->twig->addGlobal("title", "Register");
+                $this->twig->display("twigs/auth/register", $this->data);
+            }
+        } else {
+            $this->data['message'] = (validation_errors() ? validation_errors() : ($this->ion_auth->errors() ? $this->ion_auth->errors() : $this->session->flashdata('message')));
```
Comment (vibby) [inline application/controllers/User.php]: Those 4 lines of code are same than few lines earlier. Can you please find a way among ```if … then``` to use this code only once. Thank you !

**Comment by bogdanarizancu** [comment]
Hi, you are absolutely right in your observations, I think I improved everything as requested, please have in mind that the actual email sending was not tested with smtp, I did it as as I saw someone did for the forgot password functionality, I thought this was how he handled emails for now, i.e. seeing them in the browser until updating the code to actually send them. 
However, I took out the email rendering in the browser for the activation template I created as you requested, but I didn't change the forgot password "echo $content; die();" as this was not my code. Also, I changed $this->data to $data, I agree it is better, but there are many more cases of this in the controller, I did not want to interfere with other's coding, so they are still there. Please let me know if there is anything else I can do. 
Also, Michael told I should alway create a pull request for any new work, the way I did this time was to create a new feature branch and commit and push from there. I am kind of new to working with remote large teams on github, could you please shortly describe to me how should I do this next time? Thank you very much for your patience and understanding. 
All the best, Bogdan

**Comment by vibby** [comment]
Hello @bogdanarizancu. Thank you for pointing the die on forgot password stuff, I discover surprise every day too :(
I think I’m gonna accept it now, I just do a last pass on it. 
I want to thank you for your work. I know it is not easy to start as a distant worker, and there are many things we cannot share being far, things useful for project. But on my own, you do that very well !
At the the current state of the project, I think @davrv93 and michael must take a great decision about how we go from now. Keep in touch to stay tuned :) Do you have experience on apis, symfony, js frontend frameworks like react or angular ?

**Comment by bogdanarizancu** [comment]
Hi, thank you for your kind words. As for my work skills, I have worked with Laravel (not really Symphony, I know :) ) and some API interactions, but no experience with any js framework. However, I have recently opened an account on Laracasts, and Vue or React are on my next things to learn list, so let me know how I can help. Best, Bogdan

**Pull request merged**